lib/ex_doc/formatter/html/templates.ex

```diff
@@ -14,6 +14,15 @@ defmodule ExDoc.Formatter.HTML.Templates do
                     modules, exceptions, protocols)
   end
 
+  @doc """
+  Generate content for the api-reference page, provided the `modules`,
+  `exceptions` and `protocols`.
+  """
+  def api_reference_page(title, modules, exceptions, protocols, config) do
```

Comment:
module_page is called by Formatter.HTML. We should not add a function if its purpose is to be called by tests. Let's make this easier: make synopsis public and just call it directly with strings. No need to add fixtures. We will need to make synopsis public anyway when we support EPUB.